packages/hint/src/lib/config.ts

```diff
@@ -253,17 +252,55 @@ export class Configuration {
         return browserslist(config.browserslist);
     }
 
+    private static groupHintConfigs(configs: ConfigFiles, file: string): ConfigFiles {
+        const basename = path.basename(file) as ConfigFile;
+
+        if (!basename) {
+            throw new Error(`unrecognised config file '${basename}'`);
```

Comment:
The problem is if I remove the 'as ConfigFile' constraint, I get an error on line 282 'Element implicitly has an 'any' type because type 'BrowsersListConfigs' has no index signature.' I've removed the constraint and instead placed an index signature on the BrowsersListConfigs type.